packages/utils/telemetry-utils/src/errorLogging.ts

```diff
@@ -49,56 +58,112 @@ export function extractLogSafeErrorProperties(error: any) {
     return safeProps;
 }
 
-/** Type guard for ILoggingError */
-export const isILoggingError = (x: any): x is ILoggingError => typeof x?.getTelemetryProperties === "function";
+/** Metadata to annotate an error object when annotating or normalizing it */
+export interface FluidErrorAnnotations {
+    /** Telemetry props to log with the error */
+    props?: ITelemetryProperties;
+    /** fluidErrorCode to mention if error isn't already an IFluidErrorBase */
+    errorCodeIfNone?: string;
+}
+
+/** For backwards compatibility with pre-fluidErrorCode valid errors */
+function patchWithErrorCode(
+    legacyError: Omit<IFluidErrorBase, "fluidErrorCode">,
+    errorCode: string = "<error predates fluidErrorCode>",
+): asserts legacyError is IFluidErrorBase {
+    const patchMe: { fluidErrorCode?: string } = legacyError as any;
+    if (patchMe.fluidErrorCode === undefined) {
+        patchMe.fluidErrorCode = errorCode;
+    }
+}
 
 /**
- * Read-Write Logging Error.  Not exported.
- * This type alias includes addTelemetryProperties, and applies to objects even if not instanceof LoggingError\
+ * Normalize the given error yielding a valid Fluid Error
+ * @returns A valid Fluid Error with any provided annotations applied
+ * @param error - The error to normalize
+ * @param annotations - Annotations to apply to the normalized error
  */
-type RwLoggingError = LoggingError;
+export function normalizeError(
+    error: unknown,
+    annotations: FluidErrorAnnotations = {},
+): IFluidErrorBase {
+    // Back-compat, while IFluidErrorBase is rolled out
+    if (isValidLegacyError(error)) {
+        patchWithErrorCode(error, annotations.errorCodeIfNone);
+    }
 
-/** Type guard for RwLoggingError.  Not exported. */
-const isRwLoggingError = (x: any): x is RwLoggingError =>
-    typeof x?.addTelemetryProperties === "function" && isILoggingError(x);
+    if (isFluidError(error)) {
+        // We can simply annotate the error and return it
+        mixinTelemetryPropsWithFluidError(error, annotations.props);
```

Comment:
i don't understand is why IErrorBase and ILoggingError are different types, or at least why IErrorBase doesn't just extend ILoggingError. Then you could add an internal class for ErrorBase that just has get and set telemetry properites, and the mixin can just be gon